_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,240 @@
+---
+id: efbfc7
+name: Text content that updates automatically can be paused, stopped or hidden
+rule_type: atomic
+description: |
+  This rule checks that there are instruments to pause, stop or hide the auto-updating of text content.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:2.2.2: # Pause, Stop, Hide (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [visible][] [HTML element][] if:
+
+- **(changed)** the `innerText` property of the [element][html element] changes; and
```

Comment:
This rule is still making a few assumptions that aren't accounted for:

- Just because an element is visible, doesn't mean the text is. For example you can have an element with a background image and text content that is clipped in some way. I saw that on a button a little while back. You can probably capture this in the applicability somehow. Perhaps just say the element has a visible text node?

- The rule is still assuming that any innerText change is going to repeat. I don't think that's true. Most JS frameworks set up their content after load. This would capture all of those events. Another drawback of the activation notification idea is that it only works on initial page loads. You can't test SPAs with this. It also has a pretty limited set of events, so you might accidentally miss things.

What I was thinking is doing something like this:

```suggestion
- **changed:** The `innerText` property of the [element][html element] changes multiple times within a 10 minute time span where there is no user interaction; and
```

You'll have to define user interaction. I think if we define that as any event that has isTrusted set to true. That would include most assistive technologies, perhaps except for browser extensions. We should have an assumption on that. A note should also be added about the 10 minutes. It's a completely arbitrary cut-off, we can just say that anything that updates less frequently, or something that doesn't update on a regular interval may still not satisfy the SC, but that testers generally don't wait hours to see if a page changes.

Lastly, can you update the style of these labelled lists? My suggestion above is using the format also used in WCAG. I think that's the way to go on these.